packages/turf-constrained-tin/index.js

```diff
@@ -0,0 +1,897 @@
+// A fast algorithm for generating constrained delaunay triangulations
+// https://www.sciencedirect.com/science/article/pii/004579499390239A
+// A robust efficient algorithm for point location in triangulations
+// https://www.cl.cam.ac.uk/techreports/UCAM-CL-TR-728.pdf
+// https://savithru-j.github.io/cdt-js/
+// Copyright 2018 Savithru Jayasinghe
+// Licensed under the MIT License
+var helper = require('@turf/helpers');
+
+/**
+ * Takes a set of {@link Point|points} and creates a
+ * constrained [Triangulated Irregular Network](http://en.wikipedia.org/wiki/Triangulated_irregular_network),
+ * or a TIN for short, returned as a collection of Polygons. These are often used
+ * for developing elevation contour maps or stepped heat visualizations.
+ *
+ * If an optional z-value property is provided then it is added as properties called `a`, `b`,
+ * and `c` representing its value at each of the points that represent the corners of the
+ * triangle.
+ *
+ * @name constrainedTin
+ * @param {FeatureCollection<Point>} points input points
+ * @param {Array<Array<number>>} [edges] list of edges
+ * @param {String} [z] name of the property from which to pull z values
+ * This is optional: if not given, then there will be no extra data added to the derived triangles.
+ * @returns {FeatureCollection<Polygon>} TIN output
+ * @example
+ * // generate some random point data
+ * var points = turf.randomPoint(30, {bbox: [50, 30, 70, 50]});
+ *
+ * // add a random property to each point between 0 and 9
+ * for (var i = 0; i < points.features.length; i++) {
+ *   points.features[i].properties.z = ~~(Math.random() * 9);
+ * }
+ * var tin = turf.constrainedTin(points, [], 'z');
+ *
+ * //addToMap
+ * var addToMap = [tin, points]
+ * for (var i = 0; i < tin.features.length; i++) {
+ *   var properties  = tin.features[i].properties;
+ *   properties.fill = '#' + properties.a + properties.b + properties.c;
+ * }
+ */
+module.exports = function(points, edges, z) {
+    var isPointZ = false;
+    var ret = {
```

Comment:
Maybe you could find a more descriptive name for this `ret` variable?

Reply:
I will change it to "meshData"

Reply:
Done